code/modules/genetics/moecubes_printer.dm

```diff
@@ -0,0 +1,182 @@
+	var/color_key = "yellow"
+	var/obj/item/computer_hardware/hard_drive/portable/usb
+	var/list/files[0]
+	var/index = 0
+	var/gene_cache = list(
+		"name" = "\[NOT SELECTED\]",
+		"desc" = "\[NOT FOUND\]",
+		"type" = "b_type",
+		"content" = "O-")
+
+
+/obj/machinery/moeballs_printer/initalize_statverbs()
+	if(req_access.len)
+		add_statverb(/datum/statverb/hack_console)
+
+
+/obj/machinery/moeballs_printer/Initialize(mapload, d=0)
+	. = ..()
+	color_key = default_dna_machinery_style
+	update_icon()
+
+
+/obj/machinery/moeballs_printer/New()
+	..()
+	color_key = default_dna_machinery_style
+	update_icon()
+
+
+/obj/machinery/moeballs_printer/update_icon()
+	..()
+	overlays.Cut()
+	if(!(stat & (NOPOWER|BROKEN)) && use_power)
+		var/state = "printer_idle_[color_key]"
+		overlays += state
+
+
+/obj/machinery/moeballs_printer/proc/do_flick(is_error)
+	if(is_error)
+		flick("printer_error", src)
+	else
+		flick("printer_on_[color_key]", src)
+
+
+/obj/machinery/moeballs_printer/AltClick(mob/user)
+	try_eject_usb(user)
+
+
+/obj/machinery/moeballs_printer/verb/eject_usb()
+	set name = "Eject Portable Storage"
+	set category = "Object"
+	set src in view(1)
+	try_eject_usb(usr)
+
+
+/obj/machinery/moeballs_printer/proc/try_eject_usb(mob/user)
+	if(usb && eject_item(usb, user))
+		usb = null
+		files.Cut()
+		index = 0
+
+/obj/machinery/moeballs_printer/attack_hand(mob/user)
+	if(hacked || allowed(user))
+		ui_interact(user)
+	else
+		to_chat(user, SPAN_WARNING("Unauthorized access."))
+
+
+/obj/machinery/moeballs_printer/attackby(obj/item/I, mob/living/user)
+	if(istype(I, /obj/item/computer_hardware/hard_drive/portable) && !usb && insert_item(I, user))
+		usb = I
+	else if(QUALITY_PULSING in I.tool_qualities)
+		var/input_color = input(user, "Available colors", "Configuration") in GLOB.dna_machinery_styles
+		if(input_color)
+			color_key = input_color
+			update_icon()
+	else
+		..()
+
+
+/obj/machinery/moeballs_printer/emag_act(remaining_charges, mob/user, emag_source)
+	. = ..()
+	hacked = TRUE
+
+
+/obj/machinery/moeballs_printer/Topic(href, href_list)
+	if(..())
+		return TOPIC_HANDLED
+
+	if(href_list["cache"])
+		for(var/entry in files)
+			if(entry["index"] == href_list["cache"])
+				gene_cache["name"] = entry["name"]
+				gene_cache["type"] = entry["type"]
+				gene_cache["content"] = entry["content"]
+
+				switch(entry["type"])
+					if("mutation")
+						var/datum/mutation/M = entry["content"]
+						gene_cache["desc"] = M.desc
+
+					if("b_type")
+						gene_cache["desc"] = "Permanently changes blood type to [entry["content"]]."
+
+					if("real_name")
+						gene_cache["desc"] = "Alters person\'s appearance, voice and fingerprints to match those of [entry["content"]]."
+
+					if("species")
+						gene_cache["desc"] = "Significantly alters body structure. Resulting creature would clasify as [entry["content"]]"
+		return TOPIC_REFRESH
+
+	if(href_list["make_worm_cube"])
+		do_flick(FALSE)
+		sleep(1.5 SECONDS)
```

Comment:
your using both sleep and then returning topic_Refresh , whats the point ? the update will be delayed either way .

Reply:
It takes 1.5 seconds for cube printing animation to play, hence the spawning delay.

Reply:
fair